drivers/accel/adxl355/adxl355.c

```diff
@@ -0,0 +1,804 @@
+	}
+
+	status = adxl355_read_device_data(dev, ADXL355_ADDR(ADXL355_DEVID_AD),
+					  GET_ADXL355_TRANSF_LEN(ADXL355_DEVID_AD), &reg_value);
+	if ((status != 0) || (reg_value != GET_ADXL355_RESET_VAL(ADXL355_DEVID_AD)))
+		goto error;
+
+	status = adxl355_read_device_data(dev, ADXL355_ADDR(ADXL355_DEVID_MST),
+					  GET_ADXL355_TRANSF_LEN(ADXL355_DEVID_MST),&reg_value);
+	if ((status != 0) || (reg_value != GET_ADXL355_RESET_VAL(ADXL355_DEVID_MST)))
+		goto error;
+
+	status = adxl355_read_device_data(dev, ADXL355_ADDR(ADXL355_PARTID),
+					  GET_ADXL355_TRANSF_LEN(ADXL355_PARTID),&reg_value);
+	if ((status != 0) || (reg_value != GET_ADXL355_RESET_VAL(ADXL355_PARTID)))
+		goto error;
+
+	// Get shadow register values
+	status = adxl355_read_device_data(dev,
+					  ADXL355_ADDR(ADXL355_SHADOW_REGISTER_BASE_ADDR),
+					  GET_ADXL355_TRANSF_LEN(ADXL355_SHADOW_REGISTER_BASE_ADDR),
+					  &shadow_register_values[0]);
+	if (status != 0)
+		goto error;
+
+	// Default range is set
+	dev->range = ADXL355_RANGE_2G;
+
+	// Default value for FIFO SAMPLES
+	dev->fifo_samples =  ADXL355_MAX_FIFO_SAMPLES_VAL;
+
+	// Default value for POWER_CTL
+	dev->op_mode = ADXL355_STANDBY_TEMP_ON_DRDY_ON;
+
+	*device = dev;
+
+	return status;
+error:
+	free(dev);
+	return -1;
+}
+
+/***************************************************************************//**
+ * @brief Free the resources allocated by adxl355_init().
+ *
+ * @param dev 	  - The device structure.
+ *
+ * @return status - Result of the remove procedure.
+*******************************************************************************/
+int32_t adxl355_remove(struct adxl355_dev *dev)
+{
+	int32_t status = 0;
+
+	if (dev->communication_type == ADXL355_SPI_COMM)
+		status = spi_remove(dev->spi_desc);
+	else
+		status = i2c_remove(dev->i2c_desc);
+
+	free(dev);
+
+	return status;
+}
+
+/***************************************************************************//**
+ * @brief Places the device into the given operation mode.
+ *
+ * @param dev     - The device structure.
+ * @param op_mode - Operation mode mode.
+ *
+ * @return status - Result of the setting operation procedure.
+*******************************************************************************/
+int32_t adxl355_set_op_mode(struct adxl355_dev *dev,
+			    enum adxl355_op_mode op_mode)
+{
+	int32_t status = 0;
+
+	status = adxl355_write_device_data(dev, ADXL355_ADDR(ADXL355_POWER_CTL),
+					   GET_ADXL355_TRANSF_LEN(ADXL355_POWER_CTL), &op_mode);
+
+	if (status == 0)
+		dev->op_mode = op_mode;
+
+	return status;
+}
+
+/***************************************************************************//**
+ * @brief Gets the current operation mode of the device
+ *
+ * @param dev     - The device structure.
+ * @param op_mode - Read operation mode.
+ *
+ * @return status - Result of the reading operation procedure.
+*******************************************************************************/
+int32_t adxl355_get_op_mode(struct adxl355_dev *dev,
```

Comment:
Change all `status` and al functions that return a return code from `int32_t` to `int`. You were likely not aware of this, we decided at some point to just use `int` due to it being of native size for the platform (compiler takes care of that).